### DISCUSSION TITLE
Doubt related to the coalesce method

### DISCUSSION THREAD

**Discussion by yorard123** [pyg-team/pytorch_geometric#3394] [Q&A] [answered]
Hi,

I would like to know if is it possible to make this method return the 'perm' tensor that contains the new order of the 'edge_index tensor'. I am asking that, because I see that the method allows to put as input the edge_attr and the edge_indx tensors, however, I have also a tensor related to the edges called edge_label, and I would like to filter that too (it can contain duplicated entries). 

Thanks,

**Comment by wsad1** [comment, marked as answer]
You can pass both `edge_label` and `edge_attr` as a list and [torch_geometric.utils.coalesce](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.coalesce) will filter both items in the list. It would look something like this.
```
edge_index, (edge_attr, edge_label) = coalesce(edge_index, edge_attr = [edge_attr,edge_label])
```
